qcodes/instrument_drivers/Keysight/private/Keysight_344xxA.py

```diff
@@ -0,0 +1,456 @@
+from qcodes import VisaInstrument
+from pyvisa import VisaIOError
+
+log = logging.getLogger(__name__)
+
+
+class ArrayMeasurement(ArrayParameter):
+    """
+    Class to return several values. Really represents a measurement routine.
+    """
+
+    def __init__(self, name, shape=(1,), *args, **kwargs):
+
+        super().__init__(name, shape=shape, *args, **kwargs)
+
+        self.label = ''
+        self.unit = ''
+        self.properly_prepared = False
+
+    def prepare(self):
+        """
+        Prepare the measurement, create the setpoints.
+
+        There is some randomness in the measurement times.
+        """
+
+        inst = self._instrument
+
+        N = inst.sample_count()
+
+        # ensure correct instrument settings
+        inst.aperture_mode('OFF')  # aperture mode seems slower ON than OFF
+        inst.trigger_count(1)
+        inst.trigger_delay(0)
+        inst.sample_count_pretrigger(0)
+        inst.sample_source('TIM')
+        inst.autorange('OFF')
+
+        if inst.trigger_source() is None:
+            raise ValueError('Trigger source unspecified! Please set '
+                             "trigger_source to 'INT' or 'EXT'.")
+
+        # Final step
+        self.time_per_point = inst.sample_timer_minimum()
+        inst.sample_timer(self.time_per_point)
+
+        self.setpoints = (tuple(np.linspace(0, N*self.time_per_point, N)),)
+        self.shape = (N,)
+
+        self.properly_prepared = True
+
+    def get_raw(self):
+
+        if not self.properly_prepared:
+            raise ValueError('ArrayMeasurement not properly_prepared. '
+                             'Please run prepare().')
+
+        N = self._instrument.sample_count()
+        log.debug("Acquiring {} samples.".format(N))
+
+        # Ensure that the measurement doesn't time out
+        # TODO (WilliamHPNielsen): What if we wait really long for a trigger?
+        old_timeout = self._instrument.visa_handle.timeout
+        self._instrument.visa_handle.timeout = N*1000*1.2*self.time_per_point
+        self._instrument.visa_handle.timeout += old_timeout
+
+        # Turn off the display to increase measurement speed
+        self._instrument.display_text('Acquiring {} samples'.format(N))
+
+        self._instrument.init_measurement()
+        try:
+            rawvals = self._instrument.ask('FETCH?')
+        except VisaIOError:
+            rawvals = None
+            log.error('Could not pull data from DMM. Perhaps no trigger?')
+
+        self._instrument.visa_handle.timeout = old_timeout
+
+        # parse the acquired values
+        try:
+            numvals = np.array(list(map(float, rawvals.split(','))))
+        except AttributeError:
+            numvals = None
+
+        self._instrument.display_clear()
+
+        return numvals
+
+
+class _Keysight_344xxA(VisaInstrument):
+    """
+    Instrument class for Keysight 34460A, 34461A, 34465A and 34470A multimeters.
+
+    Tested with: 34461A, 34465A.
```

Comment:
is it possible to test the driver on 34460A and 34470A as well?

Reply:
No, I have only a 34461A. The 34465A was tested by the original author.
Anyway they should work, they are very similar